plugins/outputs/azuretablestorage/azuretablestorage.go

```diff
@@ -0,0 +1,259 @@
+		if !unicode.IsLetter(c) && !unicode.IsDigit(c) {
+			hex = fmt.Sprintf("%04X", utf16.Encode([]rune(string(c))))
+			encodedStr = encodedStr + replacer.Replace(hex)
+		} else {
+			encodedStr = encodedStr + string(c)
+		}
+	}
+	return encodedStr
+}
+
+//RETURNS: primary key for the azure table.
+func getPrimaryKey(resourceId string) string {
+	return encodeSpecialCharacterToUTF16(resourceId)
+}
+
+//RETURNS: difference of max value that can be held by time and number of 100 ns in current time.
+func getUTCTicks_DescendingOrder(lastSampleTimestamp string) uint64 {
+
+	currentTime, _ := time.Parse(layout, lastSampleTimestamp)
+	//maxValueDateTime := time.Date(9999, time.December, 31, 12, 59, 59, 59, time.UTC)
+	//Ticks is the number of 100 nanoseconds from zero value of date
+	//this value is copied from mdsd code.
+	maxValueDateTimeInTicks := uint64(3155378975999999999)
+
+	//zero time is taken to be 1 Jan,1970 instead of 1 Jan, 1 to avoid integer overflow.
+	//The Sub() returns int64 and hence it can hold ony nanoseconds corresponding to 290yrs.
+	zeroTime := time.Date(1970, time.January, 1, 0, 0, 0, 0, time.UTC)
+	diff := uint64(currentTime.Sub(zeroTime))
+	currentTimeInTicks := diff / 100
+	UTCTicks_DescendincurrentTimeOrder := maxValueDateTimeInTicks - currentTimeInTicks
+	fmt.Println(UTCTicks_DescendincurrentTimeOrder)
+
+	return UTCTicks_DescendincurrentTimeOrder
+}
+
+//RETURNS: counter name being unicode encoded and decreasing time diff.
+func getRowKeyComponents(lastSampleTimestamp string, counterName string) (string, string) {
+
+	UTCTicks_DescendingOrder := getUTCTicks_DescendingOrder(lastSampleTimestamp)
+	UTCTicks_DescendingOrderStr := strconv.FormatInt(int64(UTCTicks_DescendingOrder), 10)
+	encodedCounterName := encodeSpecialCharacterToUTF16(counterName)
+	return UTCTicks_DescendingOrderStr, encodedCounterName
+}
+
+func (azureTableStorage *AzureTableStorage) Write(metrics []telegraf.Metric) error {
+	var entity *storage.Entity
+	var props map[string]interface{}
+	partitionKey := getPrimaryKey(azureTableStorage.ResourceId)
+
+	// iterate over the list of metrics and create a new entity for each metrics and add to the table.
+	for i, _ := range metrics {
+		props = metrics[i].Fields()
+		UTCTicks_DescendingOrderStr, encodedCounterName := getRowKeyComponents(props[TIMESTAMP].(string), props[CounterName].(string))
+		props[DeploymentId] = azureTableStorage.DeploymentId
+		props[Host], _ = os.Hostname()
+
+		//period is the period which decides when to transfer the aggregated metrics.Its in format "60s"
+		periodStr := metrics[i].Tags()[Period]
+		table := azureTableStorage.PeriodVsTableNameVsTableRef[periodStr].TableRef
+
+		//two rows are written for each metric as Azure table has optimized prefix search only and no index.
+		rowKey1 := UTCTicks_DescendingOrderStr + "_" + encodedCounterName
+		entity = table.GetEntityReference(partitionKey, rowKey1)
+		entity.Properties = props
+		entity.Insert(FullMetadata, nil)
```

Comment:
GetEntityReference returns an empty Entity object with the values being passed to the method. As the object is created offline, it can never be null. Please add comments if i am overlooking anything.

Reply:
implementation of the GetEntityReference method is at https://github.com/Azure/azure-sdk-for-go/blob/5bf28521f7a6f72f2dde3e5a5fb0d27d7a8aeee9/storage/entity.go#L63